src/clarity_vm/tests/contracts.rs

```diff
@@ -152,3 +152,129 @@ fn test_get_burn_block_info_eval() {
+            if let Err(ClarityError::Analysis(check_error)) = res {
+                if let CheckErrors::NoSuchBlockInfoProperty(name) = check_error.err {
+                    assert_eq!(name, "block-reward");
+                } else {
+                    panic!("Bad analysis error: {:?}", &check_error);
+                }
+            } else {
+                panic!("Bad analysis result: {:?}", &res);
+            }
+        });
+    });
+    // Advance another block so we get to Stacks 2.1. This is the last block in 2.05
+    sim.execute_next_block_as_conn(|conn| {
+        let contract_identifier = QualifiedContractIdentifier::local("test-contract-2").unwrap();
+        let contract =
+            "(define-private (test-func (height uint)) (get-block-info? block-reward height))";
+        conn.as_transaction(|clarity_db| {
+            let res = clarity_db.analyze_smart_contract(&contract_identifier, contract);
+            if let Err(ClarityError::Analysis(check_error)) = res {
+                if let CheckErrors::NoSuchBlockInfoProperty(name) = check_error.err {
+                    assert_eq!(name, "block-reward");
+                } else {
+                    panic!("Bad analysis error: {:?}", &check_error);
+                }
+            } else {
+                panic!("Bad analysis result: {:?}", &res);
+            }
+        });
+    });
+    // now in Stacks 2.1, so this should work!
+    sim.execute_next_block_as_conn(|conn| {
+        let contract_identifier = QualifiedContractIdentifier::local("test-contract-3").unwrap();
+        let contract =
+            "(define-private (test-func-1 (height uint)) (get-block-info? block-reward height)) 
+             (define-private (test-func-2 (height uint)) (get-block-info? miner-spend-winner height))
+             (define-private (test-func-3 (height uint)) (get-block-info? miner-spend-total height))";
+        conn.as_transaction(|clarity_db| {
+            let (ast, _) = clarity_db
+                .analyze_smart_contract(&contract_identifier, contract)
+                .unwrap();
+            clarity_db
+                .initialize_smart_contract(&contract_identifier, &ast, contract, None, |_, _| false)
+                .unwrap();
+        });
+        let mut tx = conn.start_transaction_processing();
+        // no values for the genesis block
+        assert_eq!(
+            Value::none(),
+            tx.eval_read_only(&contract_identifier, "(test-func-1 u0)")
+                .unwrap()
+        );
+        assert_eq!(
+            Value::some(Value::UInt(0)).unwrap(),
+            tx.eval_read_only(&contract_identifier, "(test-func-2 u0)")
+                .unwrap()
+        );
+        assert_eq!(
+            Value::some(Value::UInt(0)).unwrap(),
+            tx.eval_read_only(&contract_identifier, "(test-func-3 u0)")
+                .unwrap()
+        );
+        // only works at the first block 
```

Comment:
does this really only work at the "first" block? isn't it any block that has gone by? and haven't we passed more than one?

i think it would could be good to test a subsequent block anyway, so you are testing like [0,1,N] working cases.